fleet_adapter_mir/fleet_adapter_mir/robot_adapter_mir.py

```diff
@@ -0,0 +1,767 @@
+            with mission.mutex:
+                if (mission.mission_queue_id is not None and
+                        not mission.do_not_cancel):
+                    self.api.mission_queue_id_delete(mission.mission_queue_id)
+
+    def localize(self, estimate, execution):
+        self.mission = MissionHandle(execution, localize=True)
+        self.mission.do_not_cancel = True
+        self.request_localize(estimate, self.mission)
+
+    @parallel
+    def request_localize(self, estimate, mission: MissionHandle):
+        retry_count = 10
+        count = 0
+        mission_queue_id = None
+        while count < retry_count and not mission_queue_id:
+            count += 1
+
+            # If this localize request is being made for a map switch, and a
+            # MiR position is provided in the fleet config for this RMF lift
+            # waypoint, we can localize to that position directly
+            mir_position = None
+            if estimate.inside_lift:
+                lift_name = estimate.inside_lift.name
+                this_lift = self.lift_positions.get(lift_name)
+                if this_lift and this_lift.get(estimate.map):
+                    mir_position = this_lift[estimate.map].get('name')
+
+            try:
+                mission_queue_id = self.api.localize(
+                    estimate.map, estimate.position, estimate.graph_index,
+                    mir_position
+                )
+                if mission_queue_id is not None:
+                    self.node.get_logger().info(
+                        f'[{self.name}] Localize mission is successfully '
+                        f'queued!')
+                    break
+            except Exception as err:
+                self.node.get_logger().info(
+                    f'Failed to localize on map {estimate.map}: {err}. '
+                    f'Retrying...'
+                )
+            time.sleep(1)
+
+        if not mission_queue_id:
+            self.node.get_logger().info(
+                f'[{self.name}] Failed to localize on map {estimate.map}. '
+                f'Maximum localize retries exceeded!'
+            )
+            mission.execution.finished()
+            mission_queue_id = None
+            return
+
+        mission.set_mission_queue_id(mission_queue_id)
+
+    def perform_action(self, category, description, execution):
+        if self.current_action:
+            # Should not reach here, but we log an error anyway
+            self.node.get_logger().info(
+                f'Robot [{self.name}] is busy with another perform action! '
+                f'Ignoring new action [{category}]')
+            execution.finished()
+            return
+
+        for plugin_name, config in self.plugin_config.items():
+            actions = config['actions']
+            if category in actions:
+                # Import relevant plugin
+                module = config['module']
+                plugin = importlib.import_module(module)
```

Comment:
It seems that we're loading the plugin each time we're asked to perform an action. This has a few problems:
* Loading a module with importlib is a substantial operation that involves parsing and processing the Python code of the entire module that's being loaded, including all of its imports, transitively. This is an expensive operation to be doing repeatedly.
* If the plugin is not configured correctly, we won't find out until the robot is performing a task and this function crashes.

Instead we should use importlib right when the fleet adapter is starting up. Instead of passing around a `plugin_config` dictionary that contains the config file information, we should pass around an already processed dictionary that maps from the action name to the `ActionFactory`. If there are any errors in the config file, we'll be able to catch that when we find that we're not able to create the `ActionFactory`.